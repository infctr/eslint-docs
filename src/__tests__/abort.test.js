const abort = require('../abort')
const { Abort, unabort } = abort

expect.extend({
  toResolve() {
    console.log(this, ...arguments)
  },
})

describe('abort()', () => {
  afterEach(() => {
    process.exitCode = 0
  })
  it('throws an error', () => {
    expect(abort).toThrow(Abort)
  })
  it('sets the exit code', () => {
    try {
      abort()
    } catch (e) {}

    expect(process.exitCode).toMatchSnapshot()
  })
})

const abortingFunction = () => abort()
describe('unabort', () => {
  it('wraps without error', () =>
    expect(() => {
      unabort(abortingFunction)
    }).not.toThrow(Abort))

  it('catches Abort errors', () =>
    expect(unabort(abortingFunction)()).resolves.toBe(undefined))

  it('doesnâ€™t catch non-Abort errors', () =>
    expect(
      unabort(() => {
        throw new TypeError()
      })()
    ).rejects.toBeInstanceOf(TypeError))
})
