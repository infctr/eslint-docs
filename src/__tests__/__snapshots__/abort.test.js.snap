// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abort() sets the exit code 1`] = `1`
